framework/src/modules/chain/bft/bft.js

```diff
@@ -141,6 +142,28 @@ class BFT extends EventEmitter {
 		this.finalityManager.addBlockHeader(extractBFTBlockHeaderFromBlock(block));
 	}
 
+	/**
+	 * Computes maxHeightPreviouslyForged and prevotedConfirmedUptoHeight properties that are necessary
+	 * for creating a new block
+	 * @param delegatePubKey
+	 * @return {Promise<{prevotedConfirmedUptoHeight: number, maxHeightPreviouslyForged: (number|*)}>}
+	 */
+	async computeHeadersForNewBlock(delegatePubKey) {
+		const blockHeaders = cloneDeep(this.finalityManager.headers.items);
+		const lastBlockForgedByDelegate = blockHeaders
+			.reverse()
+			.find(blockHeader => blockHeader.delegatePublicKey === delegatePubKey);
```

Comment:
Is this method of getting the lastBlockForgedByDelegate valid in all cases? Imagine a scenario where a delegate forged on a forked chain. In this case, the `lastBlockForgedByDelegate` should correspond to that block; this block will not be part of the blockHeaders.